src/textual/view.py

```diff
@@ -114,7 +114,24 @@ def get_arrangement(self, size: Size, scroll: Offset) -> Iterable[WidgetPlacemen
         cached_size, cached_scroll, arrangement = self._cached_arrangement
         if cached_size == size and cached_scroll == scroll:
             return arrangement
-        arrangement = list(self._layout.arrange(self, size, scroll))
+
+        arrangement = []
```

Comment:
I'm thinking this might simplify if we move some of the funcationality to the `WidgetPlacement` object itself.

Let's say we implement a `apply_margin` method on `WidgetPlacement` that returned either a new object or self, dependant on if there is margin or not.

Then this code could be:

```python
placements = [
    placement.apply_margin()
    for placement in self._layout.arrange(self, size, scroll)
]
```

I wouldn't be surprised if we grow other operations on the WidgetPlacement objects, and this may keep it simple.

What do you think?

Reply:
Sounds good to me. I've implemented this change now.